Accessibility improvements to site

trallard opened the pull request:
### Description

I thought it was as good time as any to give this site a bit of care 🧽  and make it a bit more accessible:

* There were several colour contrast issues. Upon closer inspection, there were too many colours used/, which made consistent use tricky (and also harder to maintain). So I simplified the colour palette, removed redundant colours, and updated the style guide.
* Links also needed a non-colour-related cue to make them more accessible and in alignment with WCAG standards. I added an underline and distinct `hover` affordances, too.
* Finally, I made some small edits to the contribution guidelines and readme


- [ ] If I have added a new page to `learn/` or `community/`, I have added it to the corresponding `_sidebar.json` file.

trallard commented on src/css/custom.css:
```diff
@@ -4,58 +4,91 @@
   --ifm-heading-font-family: nunito, sans-serif;
-  --ifm-heading-font-weight: normal;
-  --ifm-h1-font-size: 2rem;
-  --ifm-h2-font-size: 1.5rem;
-  --ifm-h3-font-size: 1.25rem;
-  --ifm-h4-font-size: 1rem;
-  --ifm-h5-font-size: 0.875rem;
-  --ifm-h6-font-size: 0.85rem;
```
These are the same as the default ones so they seemed redundant.

trallard commented on src/css/custom.css:
```diff
@@ -163,20 +196,106 @@ article {
-  color: var(--dark-theme-bg-color);
-  padding: 1rem;
-  border-radius: 5px;
+.color-table {
+  width: 90%;
+  max-width: 600px;
+  margin-top: 20px;
+  overflow: hidden;
```
Added so we could have a nice table in the style guide.

trallard commented on src/css/custom.css:
```diff
@@ -163,20 +196,106 @@ article {
+
+a:hover {
+  text-decoration: underline;
+  text-decoration-thickness: 3px !important;
+}
```
🫠 not the nicest, but had to, sorry

jaimergp commented:
Do we need underlines in _eeevery_ link? Looking at conda-forge.org, the buttons and navigation bars do not have them, and afaicr they were also compliant with WCAG. See their [custom CSS rules](https://github.com/conda-forge/conda-forge.github.io/blob/be3ddff8869ead123ab52fb5cee35e8f218767eb/src/css/custom.css#L99-L102).

jaimergp commented on docusaurus.config.js:
```diff
@@ -154,7 +154,7 @@ const config = {
           },
         ],
-        copyright: `Copyright © ${new Date().getFullYear()} Conda Community. <br /> Built with Docusaurus. This site is powered by <a href="https://www.netlify.com/" target="_blank" title="Powered by Netlify">Netlify</a>.`,
+        copyright: `Copyright © ${new Date().getFullYear()} Conda Community. <br /> Built with Docusaurus. This site is powered by <a href="https://www.netlify.com/" style="color: #d6eed1" target="_blank" title="Powered by Netlify">Netlify</a>.`,
```
Should this be slightly green or is it sufficient with just white? The rest of hyperlinks the footer and just white.

jaimergp replied:
Ok, then I guess the footer links need to be updated to match. Right now there's an inconsistency:

Normal:

<img width="319" alt="image" src="https://github.com/user-attachments/assets/23ca28a1-e4bc-4e4f-93b3-da6dd7c82d37" />

Footer hover:

<img width="277" alt="image" src="https://github.com/user-attachments/assets/11df5c7f-33be-4d6d-8543-e41d59c90a23" />


Copyright hover:

<img width="268" alt="image" src="https://github.com/user-attachments/assets/3226141c-342e-470d-bd0f-cd2960f8d4d4" />

trallard replied:
Good catch, I missed this as I had to add the colour change in place.

trallard commented:
> Do we need underlines in eeevery link? Looking at conda-forge.org, the buttons and navigation bars do not have them, and afaicr they were also compliant with WCAG. See their [custom CSS rules](https://github.com/conda-forge/conda-forge.github.io/blob/be3ddff8869ead123ab52fb5cee35e8f218767eb/src/css/custom.css#L99-L102).

That depends. We do not _need_ default underlines in the navbar as long as there are at least underlines on hover. Currently, there are none, so we rely solely on colour. 

We can remove the underlines with buttons, but I will need to see if the current colour vs hover colour are distinct enough (hue _and_ luminosity) or tweak those.

For example, the conda-forge.org ones do have a significant change in hue + luminosity on hover (white vs grey)
![conda-forge___community-driven_packaging_for_conda](https://github.com/user-attachments/assets/a16c266c-e6a3-4253-a767-501cc4ecd060)

Adding a general underline is the quickest way to add non-colour-only affordances for accessibility purposes and move towards conformance with WCAG.
If preferred, I can also go into finer detail and tweak some components so that it is not so heavy on the underlines and I have no strong feelings either way.

trallard commented:
OK I addressed all the review comments @jaimergp . This aligns now more with what we have on the conda-forge site.
I also added a couple more improvements:

1. Fixed heading order: avoid having more than one `H1` and inconsistent increases
2. Fixed the card layout. Right now, at a certain display width, the link to the blog post and the date end up overlapping:

**Current**

![image](https://github.com/user-attachments/assets/a4d69c79-b5e0-43fc-8f78-64def4926373)

**This PR ✨ **

![image](https://github.com/user-attachments/assets/670465bd-720a-499e-bd23-7d5eb4e370a0)
![conda_org___conda_org_-_Polypane](https://github.com/user-attachments/assets/4d40f73c-afd5-4ee6-be64-b1fbe7210397)

jaimergp commented:
Awesome! Looks great. I was reviewing the site now and found a couple of potential improvements:

- The sidebar for the documentation has underlines on hover now, but I don't think we need them because the background color also reacts on hover. Also not sure if we need an active state style change there too; right now I don't see any.

<img width="327" alt="image" src="https://github.com/user-attachments/assets/036d979d-f4aa-4387-a202-ccfeeac43fe9" />


- The highlighted news card hyperlink is almost black. Does that have sufficient contrast against the gradient background? I would have expected a white-ish one there, like the news header.

<img width="547" alt="image" src="https://github.com/user-attachments/assets/68703a14-01d4-4b1d-984b-7f47ec4f5ddc" />


- The top-bar conda logo does not react to hover or active states. The anvil on conda-forge.org does have some subtle size change to give a hint.
<img width="294" alt="image" src="https://github.com/user-attachments/assets/0781885a-06f6-4e96-af04-92cad7ea9e5f" />


Thanks!

trallard commented:
* I will check the sidebar states.

> The highlighted news card hyperlink is almost black. Does that have sufficient contrast against the gradient background? I would have expected a white-ish one there, like the news header.

This has a `7.7:1` contrast ✅  white on green has a `2.7:1` ⛔ 
Actually, let me check the news header again 🤔 but my last check did not throw more issues, maybe I missed something.

> The top-bar conda logo does not react to hover or active states. The anvil on conda-forge.org does have some subtle size change to give a hint.

I did not even realise, I removed some redundant animations so maybe I removed it then. Checking

jaimergp commented:
Thank you @trallard, looks fantastic!
